### PR TITLE
Using GOPROXY environment variables if set instead of default value

### PR DESCRIPTION
**Pull request by SataQiu**
### Ⅰ. Describe what this PR did
Using GOPROXY environment variables if set instead of default value

### CODE REVIEW

**Comment by Starnop** [comment]
LGTM.  cc/ @lowzj

**Comment by Starnop** [comment]
Could you please rebase your code and push it again.

**Comment by yeya24** [comment]
Just now I found we don' t have any docs about GOPROXY. Do you think we should add some about it? @SataQiu  @Starnop

**Comment by SataQiu** [comment]
> 
> 
> Just now I found we don' t have any docs about GOPROXY. Do you think we should add some about it? @SataQiu @Starnop

That's a good idea, but what to add and where to add it?

**Comment by SataQiu** [comment]
Rebased. Emmm... I wonder why rebase is needed even if there is no conflict. @Starnop

**Comment by yeya24** [comment]
> > Just now I found we don' t have any docs about GOPROXY. Do you think we should add some about it? @SataQiu @Starnop
> 
> That's a good idea, but what to add and where to add it?

Maybe to CONTRIBUTING.md? The default GOPROXY value via Makefile, and you can overwrite it by setting GOPROXY

**Comment by Starnop** [comment]
> Rebased. Emmm... I wonder why rebase is needed even if there is no conflict. @Starnop

FYI https://www.liaoxuefeng.com/wiki/896043488029600/1216289527823648

**Comment by Starnop** [comment]
> > > Just now I found we don' t have any docs about GOPROXY. Do you think we should add some about it? @SataQiu @Starnop
> > 
> > 
> > That's a good idea, but what to add and where to add it?
> 
> Maybe to CONTRIBUTING.md? The default GOPROXY value via Makefile, and you can overwrite it by setting GOPROXY

Agree. BTW. The script `hack/env.sh` will also set values for `GO111MODULE` and `GOPROXY`. Could you please make a unified？

**Comment by SataQiu** [comment]
> 
> 
> > Rebased. Emmm... I wonder why rebase is needed even if there is no conflict. @Starnop
> 
> FYI https://www.liaoxuefeng.com/wiki/896043488029600/1216289527823648

I think it would be better to use the robot to rebase automatically.

File: hack/env.sh [hunk 1]
```diff
@@ -30,4 +30,8 @@ export GOOS
 export GOARCH
 
 export GO111MODULE=on
-export GOPROXY=https://goproxy.io
+
```
Comment (Starnop) [inline hack/env.sh]: I'm wondering that whether we need `GOPROXY` both are defined in the Makefile and env.sh.

**Comment by Starnop** [comment]
> > > Rebased. Emmm... I wonder why rebase is needed even if there is no conflict. @Starnop
> > 
> > 
> > FYI https://www.liaoxuefeng.com/wiki/896043488029600/1216289527823648
> 
> I think it would be better to use the robot to rebase automatically.

In fact, it's difficult because it is a branch belongs to the user‘s own.

File: hack/env.sh (hunk 1 above)
Comment (SataQiu) [reply hack/env.sh]: /cc @allencloud

**Comment by SataQiu** [comment]
Maybe we can merge this now.
I don't think we need to make a special statement about this change because it's a development detail.
@allencloud @Starnop @yeya24

**Comment by yeya24** [comment]
LGTM @Starnop

File: hack/env.sh (hunk 1 above)
Comment (Starnop) [reply hack/env.sh]: Any updates? @SataQiu

**Comment by Starnop** [comment]
LGTM.

**Pull request merged**